### ISSUE TITLE
Modify build.xml to include/exclude CB classes and JARs

### ISSUE THREAD

**Issue by TatianaBurek** [NCAR/METviewer#96]

**Comment by JohnHalleyGotway** [comment]
Will discuss on 11/21/2018 with feedback from Randy P.

**Comment by randytpierce** [comment]
I have this working in my feature branch now. We need to discuss it before I check it in.

**Comment by randytpierce** [comment]
The idea I am working on is to use three possible -D properties to define build options.
1) -Ddb.management.system=mysql or =cb or =mariadb etc.
will override the db.management.system property file to define the system property
2) -Dbuild.property.file=propertyFileName will define a custom property file that will provide values that are in the mvservlet.properties entries.
3) -Dlog4j2.file=fullpath/customlog4j2.xml allows a builder to define a custom log24j file.

Everything should default to the way it has been if options are left out.

**Comment by randytpierce** [comment]
I added a .gitignore to the classes directory that includes only the desired basic properties (plus two .orig files) that contain no special credentials, and expressly excludes any other files. This allows the inclusion of property files that won't ever get published to git.

**Comment by randytpierce** [comment]
These are Tatiana's comments so far. Some are addressed and others are not. I am putting them here so I don't lose them.
1. Please remove from GitHub any files or file properties that contain any user names, passwords, urls, pathes, server names. We should not put any of this info into the public repository.
2. Is the new script backcompatable? Can I build  the app without any arguments and have the same result I get now? I also need flexibility on automatic deployment. I can't use ant on EMC servers so I build the app here and scp files to EMC.
3. <dist> target names a new jar file as 'metviewer-${db.management.system}.jar' but all the executable scripts (mv_batch.sh,  for example) are expecting metviewer.jar. Could you preserve the original names for the jar files and create a new attribute like 'Management system" in MANIFEST.MF instead?
4. <war> target changes the parameters in mvservlet.properties but what about other files? There are custom parameters in executable scripts and R scripts.
5. Would it make sense to create a new packages edu.ucar.metviewer.db.mysql and edu.ucar.metviewer.db.cb and move Mysql* classes to mysql package and CB* to cb package. This way the <compile> target can be simplified.

**Comment by randytpierce** [comment]
from above:
1) the .gitignore seems to accomplish this.
2) I believe that this is true, we should test it of course.
3) need to investigate.
4) need to investigate, so far I have added a test on MV_HOME such that if it is defined in the environment it will take the environment value, otherwise it leaves it the way it was before.
5) I think it is half dozen or six either way. The two packages would only have three classes each and we would really end up with three packages.. db for the base classes - DatabaseInfo, DatabaseManager, AppDataBaseManager, and LoadDatabaseManager,
db.mysql for the subclasses MysqlAppDatabaseManager, MysqlDatabaseManager, MysqlLoadDatabaseManager 
and db.cb for CBAppDatabaseManager, CBDatabaseManager, CBLoadDatabaseManager 

I think a naming convention is ok, but I don't really care either way.

Remaining work is for 3) and 4)

**Comment by randytpierce** [comment]
I have pushed my latest changes to the build.
1) make sure that the jar files and supporting database specific files are only included for the -Ddb.management.system flag that is used on the ant build command.
e.g.
ant clean;ant -Dbuild.properties.file=$HOME/IdeaProjects/METViewer/webapp/metviewer/WEB-INF/classes/buildMYSQL-docker.properties -Dlog4j2.file=$HOME/IdeaProjects/METViewer/webapp/metviewer/WEB-INF/classes/buildMYSQL-docker.log4j2.xml -Ddb.management.system=mysql -Dcontext.path.suffix="-mysql"
will build a mysql version that uses my specific build properties for a docker instance on my laptop and will append the context path "-mysql" to the build file. The flags are documented in the comments of the build.xml and I will document them further wherever the team deems necessary.
2) Take Tatianas mariadb and aurora specific changes from the feature 105 branch and incorporate them into this build scheme by making three db specific classes for mariadb and aurora databases in the same way that we did for couchbase and mysql.
3) check that the build includes only the required jar files and property files
4) make sure that any files in the WEB-INF/classes directory that are used for developing specific database connections are ignored with a gitignore entry. Make sure that any property or log files in that directory that begin with a CB, MYSQL, MARIADB, or AURORA are not included in the war or tar files.
5) refactor the build.xml a little to make the exclusions a little more concise.
6) add an environment check for the installed JAVA to the bin sh scripts so that different javas can be used and also different MV_HOME.
7) I have tested for MYSQL and CB but not mariadb or aurora.
8) I think this build stuff is ready for merging now, although I think it lacks proper documentation.
